GFDL_tools/fv_ada_nudge.F90

```diff
@@ -1793,26 +1782,35 @@ subroutine fv_ada_nudge_init(time, axes, npz, zvir, ak, bk, ts, phis, gridstruct
                'da ps', 'Pa', missing_value=missing_value )
 
 ! snz add the following lines for recording the return values from the previous assim run
-    call get_mosaic_tile_file(restart_file, restart_file_instance,&
-         & .FALSE., domain)
-
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "u_adj", Atm_var%u_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "v_adj", Atm_var%v_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "t_adj", Atm_var%t_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "q_adj", Atm_var%q_adj(:,:,:), domain=domain)
-    id_restart = register_restart_field(ada_driver_restart, restart_file_instance, &
-         & "ps_adj", Atm_var%ps_adj(:,:), domain=domain)
-
-    if ( file_exist('INPUT/'//trim(restart_file_instance), domain=domain) ) then
-       if ( mpp_pe() .eq. mpp_root_pe() ) then
-          write (stdout_unit,*) 'Reading ada restart information from ', 'INPUT/'//trim(restart_file_instance)
-       end if
-       call restore_state(ada_driver_restart, DIRECTORY='INPUT')
-    end if
+
+! set dimensions for register restart
+    dim_names_4d(1) = "xaxis_1"
+    dim_names_4d(2) = "yaxis_1"
+    dim_names_4d(3) = "zaxis_1"
+    dim_names_4d(4) = "Time"
+
+    if (open_file(ada_driver_restart, restart_file, "read", domain, is_restart=.true.)
+      call register_axis(ada_driver_restart, "xaxis_1", "X")
+      call register_axis(ada_driver_restart, "yaxis_1", "Y")
+      call register_axis(ada_driver_restart, "zaxis_1", size(Atm_var%u_adj,3)
```

Comment:
I think you are missing a final parenthesis here.  Also, why is the third argument an integer here yet in the two calls above it is a character?

Reply:
The `"x"/"y"` indicate that dimension is domain decomposed in x/y. 
The `size(Atm_var%u_adj,3)` indicates the size of that dimension.